backend/models/helper/model.js

```diff
@@ -1005,10 +1010,61 @@ function getMetadata(account, model, id) {
 			return Promise.reject(responseCodes.METADATA_NOT_FOUND);
 		}
 	});
+}
+
+async function getMeshById(account, model, meshId) {
+	const historyRes =  (await History.findByObjectId(account, model, meshId, {current:1}));
+	if (!historyRes) {
+		throw responseCodes.RESOURCE_NOT_FOUND;
+	}
+
+	const revisionIds = historyRes.current;
+	const projection = {
+		"parents": 1,
+		"vertices": 1,
+		"matrix": 1,
+		"faces": 1,
+		"_extRef":1
+	};
+
+	const mesh = await Scene.getObjectById(account, model, utils.stringToUUID(meshId), projection);
+	let parentMatrix = null;
+
+	if ((mesh.parents || []).length > 0) {
+		parentMatrix = await getParentMatrix(account, model, mesh.parents[0], revisionIds);
+		if (mesh.matrix) {
```

Comment:
don't think this is needed a mesh will never have matrix in it by specification.